### ISSUE TITLE
Support Py_LIMITED_API with pyro C-API headers

### ISSUE THREAD

**Issue by tekknolagi** [tekknolagi/skybison#186]
T66752724:
The pyro C-API headers (pyro/capi/*.h) should behave like cpython and hide all declarations that are not part of PEP384 (aka limited API) when Py_LIMITED_API is not set.

Proposal for how to do this:
Create capi/cpython-func-extra.h and move everything from capi/cpython-func.h that is not part of the limited API into this new header. Then add a #ifndef PY_LIMITED_API #include "cpython-func-extra.h" #endif at the end of the file.
Occasionally you can find this construct in the cpython headers: `#if !defined(Py_LIMITED_API) || Py_LIMITED_API+0 >= 0x03030000` I think it would be fine to freeze our API for whatever is in cpython 3.7 and not provide these extra checks... That probably means we should reject compilation requests that ask for Py_LIMITED_API without version number or smaller than 3.7. (alternative to this whole fixing would be to have bigger blocks of this check in our headers...)

Created-Date: 2020/05/08 16:33:08
Blocks: T53256673
